explore/src/main/scala/ColumnSelector.scala

```diff
@@ -0,0 +1,62 @@
+// Copyright (c) 2016-2022 Association of Universities for Research in Astronomy, Inc. (AURA)
+// For license information see LICENSE or https://opensource.org/licenses/BSD-3-Clause
+
+package lucuma.ui.table
+
+import cats.syntax.all.*
+import crystal.react.View
+import explore.model.TableColumnPref
+import japgolly.scalajs.react.*
+import japgolly.scalajs.react.vdom.html_<^.*
+import react.common.*
+import react.semanticui.modules.checkbox.Checkbox
+import react.semanticui.modules.dropdown.*
+import reactST.{tanstackTableCore => raw}
+
+case class NewColumnSelector[T](
```

Comment:
Can we keep using some version of the `ColumnSelector` we have in `lucuma-ui` so that we can share it with other projects and save preferences by using `withOnMod` on the `View`?

Reply:
I was going to replace `ColumnSelector` with this one. I guess we can keep them both, Saving is now done using `withOnMod`

Reply:
A big difference is that the old selector worked on a list of hidden columns, while this works on a list of all columns, you want to know all columns to save which ones are visible and which are not ideally in one go

Reply:
Ah, I see. Ideally we should keep only one version, and have it in lucuma-ui.